catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -121,46 +173,95 @@ public void updateCallback(Map<String, Object> properties) {
             }
             init();
         }
-
-        LOGGER.trace("EXITING: updateCallback");
     }
 
     /**
      * @param monitoredDirectoryPath - directory path for the monitored directory
      */
     public void setMonitoredDirectoryPath(String monitoredDirectoryPath) {
-        LOGGER.trace("INSIDE: setMonitoredDirectoryPath");
-
         this.monitoredDirectory = monitoredDirectoryPath;
     }
 
     /**
      * @param copyIngestedFiles - flag to copy ingested files
      */
     public void setCopyIngestedFiles(boolean copyIngestedFiles) {
-        LOGGER.trace("INSIDE: setCopyIngestedFiles");
-
         this.copyIngestedFiles = copyIngestedFiles;
     }
 
+    /**
+     * @param attributeOverrides - a list of attributes to override
+     */
     public void setAttributeOverrides(List<String> attributeOverrides) {
         this.attributeOverrides = attributeOverrides;
     }
 
-    public List<String> getAttributeOverrides() {
-        return this.attributeOverrides;
+    public List<RouteDefinition> getRouteDefinitions() {
+        return camelContext.getRouteDefinitions();
     }
 
-    private void configureCamelRoute() {
-        LOGGER.trace("ENTERING: configureCamelRoute");
-
-        // Must have a directory to be monitored to be able to configure the Camel route.
-        if (StringUtils.isEmpty(monitoredDirectory)) {
-            LOGGER.debug("Cannot setup camel route - must specify a directory to be monitored");
-            return;
+    /*
+        Task that waits for the "content" CamelComponent before adding the routes from the
+        RouteBuilder to the CamelContext. This ensures content directory monitors will
+        automatically start after a system shutdown.
+     */
+    private void attemptAddRoutes() {
+        LOGGER.debug("Attempting to add routes for content directory monitor watching {}",
+                monitoredDirectory);
+        RouteBuilder routeBuilder = createRouteBuilder();
+        try {
+            verifyContentCamelComponentIsAvailable();
+            camelContext.addRoutes(routeBuilder);
+            setRouteCollection(routeBuilder);
+        } catch (InterruptedException e) {
+            Thread.currentThread()
+                    .interrupt();
```

Comment:
If this is interrupted at the very beginning do you not want to try to retry? Also do all three of these methods throw interrupted exception or do you need them for just a specific one?